### PR TITLE
[Merged by Bors] - chore: forward-port leanprover-community/mathlib#19084 (file 1 of 2)

### PR DESCRIPTION
**Pull request by Multramate**
[`linear_algebra.free_module.pid`@`210657c4ea4a4a7b234392f70a3a2a83346dfa90`..`d87199d51218d36a0a42c66c82d147b5a7ff87b3`](https://leanprover-community.github.io/mathlib-port-status/file/linear_algebra/free_module/pid?range=210657c4ea4a4a7b234392f70a3a2a83346dfa90..d87199d51218d36a0a42c66c82d147b5a7ff87b3)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/FreeModule/PID.lean [hunk 1]
```diff
@@ -624,7 +624,7 @@ theorem Ideal.smithCoeffs_ne_zero (b : Basis ι R S) (I : Ideal S) (hI : I ≠ 
 #align ideal.smith_coeffs_ne_zero Ideal.smithCoeffs_ne_zero
 
 instance (F : Type u) [CommRing F] [Algebra F R] (b : Basis ι R S) {I : Ideal S} (hI : I ≠ ⊥) (i) :
-    Module F (R ⧸ Ideal.span ({I.smithCoeffs b hI i} : Set R)) := by
+    Module F (R ⧸ Ideal.span ({I.smithCoeffs b hI i} : Set R)) :=
   Quotient.module' _
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/FreeModule/PID.lean:628]: I wasn't suggesting that you _actually_ replace this with `Quotient.module'`; just that you check the comment was accurate (it previously wasn't).

I don't really understand why we added this instance in Lean 3 in the first place though if `apply_instance` could find it there.

Comment (Multramate) [reply Mathlib/LinearAlgebra/FreeModule/PID.lean:628]: I wasn't able to do anything locally so I thought I'd let CI tell me

Comment (Multramate) [reply Mathlib/LinearAlgebra/FreeModule/PID.lean:628]: I guess since #4821 works just fine without this, I'll delete this PR

**Comment by eric-wieser** [comment]
You should add a porting note and noalign where the instance was removed

**Review by alreadydone — APPROVED** [review]

File: Mathlib/LinearAlgebra/FreeModule/PID.lean [hunk 2]
```diff
@@ -623,6 +623,9 @@ theorem Ideal.smithCoeffs_ne_zero (b : Basis ι R S) (I : Ideal S) (hI : I ≠ 
   simp [hi]
 #align ideal.smith_coeffs_ne_zero Ideal.smithCoeffs_ne_zero
 
+-- porting note: can be inferred in Lean 4 so no longer necessary
+#noalign has_quotient.quotient.module
```
Comment (alreadydone) [inline Mathlib/LinearAlgebra/FreeModule/PID.lean:627]: LGTM, though the mathlib3 name is a bit funny ...

**Review by eric-wieser — APPROVED** [review]
bors merge

Thanks!

**Pull request closed without merging**